src/status_im/ui/components/bottom_sheet/view.cljs

```diff
@@ -90,75 +83,116 @@
+            [react/animated-view (merge
+                                  {:pointer-events "box-none"
+                                   :style          (styles/content-container window-height sheet-height bottom-value)}
+                                  (when-not (or disable-drag?
+                                                (= max-height sheet-height))
+                                    (pan-handlers
+                                     (swipe-pan-responder {:bottom-value  bottom-value
+                                                           :opacity-value opacity-value
+                                                           :height        height
+                                                           :close-sheet   #(close-sheet)}))))
+             [react/view (merge {:style styles/content-header}
+                                (when-not disable-drag?
+                                  (pan-handlers
+                                   (swipe-pan-responder {:bottom-value  bottom-value
+                                                         :opacity-value opacity-value
+                                                         :height        height
+                                                         :close-sheet   #(close-sheet)}))))
+              (when show-handle?
+                [react/view styles/handle])]
+             [react/animated-view {:style {:height sheet-height}}
+              ;; NOTE: For a better UX on onScrollBeginDrag we can start dragging the sheet.
+              [react/scroll-view {:bounces        false
+                                  :scroll-enabled true
+                                  :style          {:flex 1}}
+               [react/view {:style     {:padding-top    styles/vertical-padding
+                                        :padding-bottom (+ styles/vertical-padding
+                                                           (:bottom safe-area))}
+                            :on-layout #(->> %
+                                             .-nativeEvent
+                                             .-layout
+                                             .-height
+                                             (reset! content-height))}
+                [content]]]]]]])))))
 
-                (and old-show? (false? new-show?) (true? @show-sheet?))
-                (cancel opts'))))
-      :reagent-render
-      (fn [{:keys [content content-height]}]
-        (let [total-content-height (+ content-height
-                                      styles/border-radius
-                                      styles/bottom-padding)]
-          (when @show-sheet?
-            [bottom-sheet-view (assoc opts
-                                      :content content
-                                      :height total-content-height)])))})))
+(defn bottom-sheet [props]
+  [react/safe-area-consumer
+   (fn [insets]
+     (reagent/as-element
```

Comment:
why do we need `as-element` here ?

Reply:
Because it is a react context consumer https://github.com/reagent-project/reagent/blob/ff4c9a6031712994ae92ac60fdd7df8ec77e92f8/doc/ReactFeatures.md#context

Reply:
Isn't it possible to use the safe-area-view and avoid that consumer?

Reply:
In this case no, cause I need the value to use in calculations https://github.com/status-im/status-react/pull/9915/files/96f371faf9a254f0b0fe7821ebee23819e37cbbd#diff-04c3dd84b9db26e4b7433a9e25932009R127 but yes, for cases where the only padding is needed better to use `safe-area-view`